lib/miq_ldap.rb

```diff
@@ -266,18 +266,24 @@ def normalize(dn)
     dn.split(",").collect { |i| i.downcase.strip }.join(",")
   end
 
-  def is_dn?(str)
+  def dn?(str)
     !!(str =~ /^([a-z|0-9|A-Z]+ *=[^,]+[,| ]*)+$/)
   end
 
+  def upn?(str)
+    !!(str =~ /^.+@.+$/)
+  end
+
   def domain_username?(str)
     !!(str =~ /^([a-zA-Z][a-zA-Z0-9.-]+)\\.+$/)
   end
 
   def fqusername(username)
-    return username if self.is_dn?(username) || self.domain_username?(username)
+    return username if self.dn?(username) || self.domain_username?(username)
 
     user_type = @user_type.split("-").first
+    return username if user_type != "mail" && self.upn?(username)
```

Comment:
why was this needed ?

Reply:
This was necessary because "mail" user_type can also have a "@". Similar to your comment regarding rdns in ldap.
Upon reviewing the old code, where there was logic to handle usernames with "@" when user_type was "mail" I realized this change was necessary.